src/js/select2/core.js

```diff
@@ -94,7 +94,8 @@ define([
     if ($element.attr('id') != null) {
       id = $element.attr('id');
     } else if ($element.attr('name') != null) {
-      id = $element.attr('name') + '-' + Utils.generateChars(2);
+      id = $element.attr('name').replace(/\[|\]/g, '');
```

Comment:
We also might want to apply this to the `id` case above, as it also has the same issue (though less common).